src/snowflake/snowpark/stored_procedure.py

```diff
@@ -756,6 +757,23 @@ def _do_register_sp(
             UDFColumn(dt, arg_name) for dt, arg_name in zip(input_types, arg_names[1:])
         ]
 
+        # Add in snowflake-snowpark-python if it is not already in the package list.
+        major, minor, patch = VERSION
+        package_name = "snowflake-snowpark-python"
+        # Use == to ensure that the remote version matches the local version
+        this_package = f"{package_name}=={major}.{minor}.{patch}"
+
+        # When resolve_imports_and_packages is called below it will use the provided packages or
+        # default to the packages in the current session. If snowflake-snowpark-python is not
+        # included by either of those two mechanisms then create package list does include it and
+        # any other relevant packages.
+        if packages is None:
+            if package_name not in self._session._packages:
```

Comment:
Is there a problem if `self._session._packages` has `snowflake-snowpark-python==1.9.0`?

Reply:
No. `self._session._packages` is a dictionary with keys that are the package name and values that are the specific string of the requirement. In your example it might be structured something like this:
```
{
    "snowflake-snowpark-python": "snowflake-snowpark-python==1.9.0"
}
```